src/api/objects/ObjectAPI.js

```diff
@@ -229,12 +229,10 @@ export default class ObjectAPI {
             delete this.cache[keystring];
 
             result = this.applyGetInterceptors(identifier, result);
-            if (result.isMutable) {
-                result.$refresh(result);
-            } else {
-                let mutableDomainObject = this.toMutable(result);
-                mutableDomainObject.$refresh(result);
-            }
+
+            const mutableDomainObject = this.toMutable(result);
```

Comment:
We can skip all this and save some cycles if the object does not support mutation (like a telemetry object) by first calling `this.supportsMutation(domainObject)`